gufe/tests/test_models.py

```diff
@@ -41,23 +69,135 @@ def test_default_settings():
     my_settings.schema_json(indent=2)
 
 
-@pytest.mark.parametrize(
-    "value,good",
-    [
-        ("parsnips", False),  # shouldn't be allowed
-        ("hbonds", True),
-        ("hangles", True),
-        ("allbonds", True),  # allowed options
-        ("HBonds", True),  # check case insensitivity
-    ],
-)
-def test_invalid_constraint(value, good):
-    if good:
-        s = OpenMMSystemGeneratorFFSettings(constraints=value)
-        assert s
-    else:
-        with pytest.raises(ValueError):
-            _ = OpenMMSystemGeneratorFFSettings(constraints=value)
+class TestSettingsValidation:
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            ("parsnips", False, None),  # shouldn't be allowed
+            ("hbonds", True, "hbonds"),
+            ("hangles", True, "hangles"),
+            ("allbonds", True, "allbonds"),  # allowed options
+            ("HBonds", True, "HBonds"),  # check case insensitivity TODO: cast this to lower?
```

Comment:
Good question, I think b/c settings are checked at instantiation time and normally right at the start of any workflow, _I_ think we should be more strict about things like the case of `HBonds` and platform names `CUDA` even though we could decide on `.lower()` everything. I think for 2.0 we should decide on how much we coerce things versus reject them.